pkg/controller/constraint/constraint_controller.go

```diff
@@ -151,12 +188,55 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 			if err := r.Update(context.Background(), instance); err != nil {
 				return reconcile.Result{Requeue: true}, nil
 			}
+			// removing constraint entry from cache
+			delete(r.constraintsCache, constraintKey)
 		}
 	}
-
+	r.reportTotalConstraints(
+		constraintKey,
+		Tags{
+			enforcementAction: enforcementAction,
+			status:            activeStatus,
+		})
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileConstraint) reportTotalConstraints(constraintKey string, t Tags) {
+	r.constraintsCacheMux.Lock()
+	defer r.constraintsCacheMux.Unlock()
+
+	r.constraintsCache[constraintKey] = Tags{
+		enforcementAction: t.enforcementAction,
+		status:            t.status,
+	}
+
+	totals := make(map[Tags]int)
+	// report total number of constraints
+	for _, v := range r.constraintsCache {
+		totals[v] += 1
+	}
+	if err := r.reporter.ReportConstraints(
+		Tags{enforcementAction: util.Deny, status: activeStatus},
+		int64(totals[Tags{enforcementAction: util.Deny, status: activeStatus}])); err != nil {
+		log.Error(err, "failed to report constraint")
+	}
+	if err := r.reporter.ReportConstraints(
+		Tags{enforcementAction: util.Deny, status: errorStatus},
+		int64(totals[Tags{enforcementAction: util.Deny, status: errorStatus}])); err != nil {
+		log.Error(err, "failed to report constraint")
+	}
+	if err := r.reporter.ReportConstraints(
+		Tags{enforcementAction: util.Dryrun, status: activeStatus},
+		int64(totals[Tags{enforcementAction: util.Dryrun, status: activeStatus}])); err != nil {
+		log.Error(err, "failed to report constraint")
+	}
+	if err := r.reporter.ReportConstraints(
+		Tags{enforcementAction: util.Dryrun, status: errorStatus},
+		int64(totals[Tags{enforcementAction: util.Dryrun, status: errorStatus}])); err != nil {
+		log.Error(err, "failed to report constraint")
+	}
```

Comment:
This is okay for now, but these if statements won't scale well as new tags and/or tag values get added over time. Consider switching to nested for loops at some point.

Reply:
+1 switching now would avoid bugs in the future where we add a new enforcement action but don't remember to update this portion of the code.

```
for tag, count := range totals {
   if err := r.reporter.ReportConstraints(tags, int64(totals)); err != nil { ...
}
```

Reply:
Also, it looks like we are not reporting `unknown` or `unsupported` enforcement actions (whichever name we went with).

Reply:
this doesn't work because we are removing the `constraintKey`, once we remove that and there is no more Tags of that kind left to loop (`dryrun` and `active` for example), it doesn't report `0` as the total constraint count.

Reply:
Fair. I see two options:

1) Nested for loops across the known constraint types and known statuses
2) Keep a record of which keys we've already seen and loop over that

Reply:
makes sense, updated with nested loop for now